### PR TITLE
Starting point moveit_ros_planning_interface

### PR DESCRIPTION
**Pull request by anasarrak**
@vmayoral Our work for now is to start wit the port of planning_interface, inside this, the most valuable part is the [move_group_interface](https://github.com/AcutronicRobotics/moveit2/blob/2eb7a8030b39848a1b803e916f036f853aef2c24/moveit_ros/planning_interface/CMakeLists.txt#L79), with that, we can make a minimal execution working.

the move_group_interface depends from:

1. **[Perception](https://github.com/AcutronicRobotics/moveit2/tree/moveit_ros_planning_interface/moveit_ros/perception)**
1. **[Planning](https://github.com/AcutronicRobotics/moveit2/tree/moveit_ros_planning_interface/moveit_ros/planning)**
1. **[Move_group](https://github.com/AcutronicRobotics/moveit2/tree/moveit_ros_planning_interface/moveit_ros/move_group)**
1. **[Warehouse](https://github.com/AcutronicRobotics/moveit2/tree/moveit_ros_planning_interface/moveit_ros/warehouse)**(I don't know how important is this last one, but is a needed dependencie)

### CODE REVIEW

File: moveit_ros/planning_interface/CMakeLists.txt [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
-  tf2_geometry_msgs
-  tf2_ros
-  roscpp
-  actionlib
-  rospy
-  rosconsole
-)
+find_package(ament_cmake REQUIRED)
+find_package(moveit_msgs REQUIRED)
+# find_package(moveit_ros_planning REQUIRED)
```
Comment (vmayoral) [inline moveit_ros/planning_interface/CMakeLists.txt:16]: Why is this commented right away? Is it because we're expecting to uncomment it once we port these submodules? If so, find below a suggestion:
```suggestion
# TODO: uncomment this as submodules become available
# find_package(moveit_ros_planning REQUIRED)
```

File: moveit_ros/planning_interface/CMakeLists.txt [hunk 2]
```diff
@@ -34,46 +36,35 @@ elseif()
   set(BOOST_PYTHON_COMPONENT python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
 endif()
 
-#catkin_lint: ignore_once duplicate_find
-find_package(Boost REQUIRED COMPONENTS
```
Comment (vmayoral) [inline moveit_ros/planning_interface/CMakeLists.txt]: You just removed these packages, shouldn't we do as above and require them?

File: moveit_ros/planning_interface/CMakeLists.txt [hunk 3]
```diff
@@ -34,46 +36,35 @@ elseif()
...
-
 set(THIS_PACKAGE_INCLUDE_DIRS
   py_bindings_tools/include
   common_planning_interface_objects/include
   planning_scene_interface/include
   move_group_interface/include
 )
 
-catkin_python_setup()
+# set(THIS_PACKAGE_INCLUDE_LIBRARIES
```
Comment (vmayoral) [inline moveit_ros/planning_interface/CMakeLists.txt]: Why is this here?

Comment (vmayoral) [reply moveit_ros/planning_interface/CMakeLists.txt]: Meaning what's below

File: moveit_ros/planning_interface/package.xml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<package format="2">
+<package format="3">
```
Comment (vmayoral) [inline moveit_ros/planning_interface/package.xml:3]: Let's follow official recommendations please @anasarrak
```suggestion
<?xml version="1.0"?>
<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
```

**Review by vmayoral — APPROVED** [review]

**Comment by vmayoral** [comment]
@anasarrak, have a look at the comments/suggestions above. Other than that, I'd get this and keep moving forward.

File: moveit_ros/planning_interface/CMakeLists.txt (hunk 1 above)
Comment (anasarrak) [reply moveit_ros/planning_interface/CMakeLists.txt:16]: Moveit_ros_planning depends from moveit_ros_perception, once I finish with that, ill uncomment it. I'll add the TODO:

File: moveit_ros/planning_interface/CMakeLists.txt (hunk 2 above)
Comment (anasarrak) [reply moveit_ros/planning_interface/CMakeLists.txt]: True, I just looked at the doc and we should do it as above... my bad

File: moveit_ros/planning_interface/CMakeLists.txt [hunk 5]
```diff
@@ -23,7 +24,15 @@ find_package(tf2_geometry_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclpy REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED COMPONENTS
```
Comment (anasarrak) [inline moveit_ros/planning_interface/CMakeLists.txt]: We have to leave it this way

**Comment by anasarrak** [comment]
Changes made

File: moveit_ros/planning_interface/package.xml [hunk 6]
```diff
@@ -8,34 +10,58 @@
...
+  <build_depend>rospy</build_depend>
+  <build_depend>rosconsole</build_depend>
+  <build_depend>actionlib</build_depend>
+  <build_depend>geometry_msgs</build_depend>
+  <build_depend>moveit_msgs</build_depend>
+  <build_depend>tf2</build_depend>
+  <build_depend>tf2_eigen</build_depend>
+  <build_depend>tf2_geometry_msgs</build_depend>
+  <build_depend>tf2_ros</build_depend>
+  <build_depend>python</build_depend>
```
Comment (ahcorde) [inline moveit_ros/planning_interface/package.xml]: ?

File: moveit_ros/planning_interface/package.xml [hunk 7]
```diff
@@ -8,34 +10,58 @@
...
+  <exec_depend>rospy</exec_depend>
+  <exec_depend>rosconsole</exec_depend>
+  <exec_depend>actionlib</exec_depend>
+  <exec_depend>geometry_msgs</exec_depend>
+  <exec_depend>moveit_msgs</exec_depend>
+  <exec_depend>tf2</exec_depend>
+  <exec_depend>tf2_eigen</exec_depend>
+  <exec_depend>tf2_geometry_msgs</exec_depend>
+  <exec_depend>tf2_ros</exec_depend>
+  <exec_depend>python</exec_depend>
```
Comment (ahcorde) [inline moveit_ros/planning_interface/package.xml]: ?

File: moveit_ros/planning_interface/package.xml [hunk 8]
```diff
@@ -8,34 +10,58 @@
...
+  <exec_depend>eigen</exec_depend>
```
Comment (ahcorde) [inline moveit_ros/planning_interface/package.xml]: is there any package in ROS 2.0 called `python` or `eigen`. Otherwise you need to remove these lines

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by vmayoral** [comment]
> Changes made

Thanks for this but you could literally have saved you some time by accepting the suggestions via the web editor (so that you don't need to apply them manually ;) ).

File: moveit_ros/planning_interface/CMakeLists.txt (hunk 5 above)
Comment (vmayoral) [reply moveit_ros/planning_interface/CMakeLists.txt]: I doubt we need to use the `COMPONENTS` keyword. Can you justify why?

**Comment by vmayoral** [comment]
As usual, it seems that we've missed some dependencies _for those of us building from source_:
```bash
/Users/victor/ros2_moveit_ws/src/moveit2/moveit_core/dynamics_solver/include/moveit/dynamics_solver/dynamics_solver.h:41:10: fatal error: 'kdl/chain.hpp' file not found
#include <kdl/chain.hpp>
         ^~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [dynamics_solver/CMakeFiles/moveit_dynamics_solver.dir/src/dynamics_solver.cpp.o] Error 1
make[1]: *** [dynamics_solver/CMakeFiles/moveit_dynamics_solver.dir/all] Error 2
```

**Comment by vmayoral** [comment]
Fixed the issues above with https://github.com/AcutronicRobotics/moveit2/pull/23/commits/bc616b2f16b7c3a8d993abd9a19ad4b483a48348

File: moveit_ros/planning_interface/package.xml (hunk 8 above)
Comment (vmayoral) [reply moveit_ros/planning_interface/package.xml]: @anasarrak can you please address these concerns?

File: moveit_ros/planning_interface/CMakeLists.txt (hunk 5 above)
Comment (anasarrak) [reply moveit_ros/planning_interface/CMakeLists.txt]: We just have to use ``find_package(Boost REQUIRED)`` we cannot split the components into ``find_package(date_time REQUIRED)``

File: moveit_ros/planning_interface/package.xml (hunk 8 above)
Comment (anasarrak) [reply moveit_ros/planning_interface/package.xml]: Changes already made https://github.com/AcutronicRobotics/moveit2/pull/23/commits/34d992b346010d05a37b8158c503decaaa5e238e

**Comment by vmayoral** [comment]
@anasarrak I've fixed the conflicts and tested locally whether it builds. Seems things build nicely so I'll accept this. 

Advice though, please keep PRs package specific. You are mixing things here between moveit_core and moveit_ros_planning_interface. Submitting PRs upstream will become more work if we don't keep things separated.
